### PR TITLE
Initial queryparser

### PR DESCRIPTION
**Pull request by ISOsipov**
# Filter
## Version number 1
This is the first version of our filter. Now it can allow us to find rows using:
- Comparators: _<, >, =, >=, <=, !=_ ;
- Logical operators: _or, and_ .

For example, we can input the next string for filter:

_variable1 >= number and variable2 != 'string' or variable3 = variable4_

### CODE REVIEW

**Comment by CimimUxMaio** [comment]
I've uploaded some tests for this functionality.

To run them you have to use the command:

```
python -m unittest
```

Currently they are all failing due to the same error:
image
 
Can you investigate? You may want to look into the tests code, maybe I am using this function wrong.

File: src/queryparser.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+def parse(filter: str, varList: list):
```
Comment (CimimUxMaio) [inline src/queryparser.py]: To be consistent with the module's name I would rename this variable to `query`.

Comment (CimimUxMaio) [inline src/queryparser.py]: `varList` should be `var_list` (snake_case) to be consistent with coding style.

File: src/queryparser.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    len_varList = len(varList)
+    n = len_varList
+    adaptetd_var_list = []
```
Comment (CimimUxMaio) [inline src/queryparser.py]: Typo - I assume "adaptetd" should be "adapted".

Comment (ISOsipov) [reply src/queryparser.py]: Thank you, it was fixed

File: src/queryparser.py (hunk 1 above)
Comment (ISOsipov) [reply src/queryparser.py]: Thank you, it was fixed

Comment (ISOsipov) [reply src/queryparser.py]: Thank you, it was changed

**Pull request merged**